### PR TITLE
Avoid using Duration.parse directly due to JDK8 bug

### LINKED ISSUE #214: Cannot rely on JDK8 Duration.parse
Due to a [bug](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8054978), we may not rely on this method for parsing durations from strings, and should manually implement a workaround.

### PR DESCRIPTION
**Pull request by calvinlfer**
Ported the JDK16 implementation of `Duration.parse` to work around the JDK8 issue described https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8054978

Fixes https://github.com/zio/zio-json/issues/214

### CODE REVIEW

**Comment by fsvehla** [comment]
CI is upset. Did you run `prepare`?

**Comment by calvinlfer** [comment]
oops let me fix

File: zio-json/shared/src/main/scala/zio/json/decoder.scala [hunk 1]
```diff
@@ -676,6 +676,128 @@ private[json] trait DecoderLowPriority3 {
     }
 }
 
+// The code in JavaTimeWorkaround is a more Scala-friendly port of the Duration.parse method from JDK 16 to cope with
+// a bug resulting in incorrect durations in JDK 8 (see https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8054978)
+private[json] object JavaTimeWorkaround {
```
Comment (fsvehla) [inline zio-json/shared/src/main/scala/zio/json/decoder.scala:681]: Nitpicking: I’d move this to a new file, maybe in an `util` package nested under the package of this class

Comment (calvinlfer) [reply zio-json/shared/src/main/scala/zio/json/decoder.scala:681]: Will make the change!

**Comment by calvinlfer** [comment]
@fsvehla should I remove the Either and just throw exceptions to avoid the extra overhead (relying on `mapStringOrFail` to catch Exceptions)

**Comment by fsvehla** [comment]
@calvinlfer Sure — why not follow up on this one?

**Comment by fsvehla** [comment]
@calvinlfer With both of these changes

**Pull request merged**